modules/phase_field/include/materials/EpsAndMAnisoGauss.h

```diff
@@ -0,0 +1,143 @@
+// 5D Gaussian anisotropy material object
+// Material properties adding anisotropy to epsilon [or also called k] and m;
+// L is isotropic and computed from a given mobility.
+
+#pragma once
+
+#include "ADMaterial.h"
+#include "MooseMesh.h"
+#include "MathUtils.h"
+#include "GrainTrackerInterface.h"
+#include <vector>
+
+class EpsAndMAnisoGauss : public ADMaterial
+{
+public:
+  static InputParameters validParams();
+
+  EpsAndMAnisoGauss(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties() override;
+
+private:
+  void readQuaternionFile();
+  void readLibraryFile();
+
+  // Grains orientation quaternions.
+  std::vector<double> _qwR, _qxR, _qyR, _qzR;
+  // Normalized_Axis_X  Normalized_Axis_Y Normalized_Axis_Z  Angle(Radian) GB_Normal_X  GB_Normal_Y
+  // GB_Normal_Z  Minimun_GB_energy(J/m^2).
+  std::vector<double> _vbalibx, _vbaliby, _vbalibz, _thetabalib, _miubalibx, _miubaliby, _miubalibz,
+      _MinimaEnergy;
```

Comment:
I just saw that line in the file above. Please think about whether you can factor out a common base class.

Reply:
I ended merging all four materials into a single file